spinnaker/spinnaker_system/google_http_lb_test.py

```diff
@@ -0,0 +1,1256 @@
+       [
+         jp.PathContainsPredicate('name', 'backend-service'),
+         jp.PathContainsPredicate('port',
+                                  int(self.__default_upsert[0]['portRange'])),
+         jp.PathContainsPredicate('healthChecks[0]', hc_link),
+       ]
+     ).contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', 'default-backend-service'),
+         jp.PathContainsPredicate('port',
+                                  int(self.__default_upsert[0]['portRange'])),
+         jp.PathContainsPredicate('healthChecks[0]', hc_link),
+       ]
+     ).contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', 'pm-backend-service'),
+         jp.PathContainsPredicate('port',
+                                  int(self.__default_upsert[0]['portRange'])),
+         jp.PathContainsPredicate('healthChecks[0]', hc_link),
+       ]
+     )
+    )
+    (contract_builder.new_clause_builder('Url Map Created',
+                                        retryable_for_secs=30)
+     .list_resource('urlMaps')
+     .contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', self.__lb_name),
+         jp.PathContainsPredicate('defaultService',
+                                  self._make_bs_link('default-backend-service')),
+         jp.PathContainsPredicate('pathMatchers/defaultService',
+                                  self._make_bs_link('pm-backend-service')),
+         jp.PathContainsPredicate('pathMatchers/pathRules/service',
+                                  self._make_bs_link('backend-service')),
+         jp.PathContainsPredicate('pathMatchers/pathRules/paths', '/path'),
+         jp.PathContainsPredicate('pathMatchers/pathRules/paths', '/path2/more'),
+         jp.PathContainsPredicate('hostRules/hosts', 'host1.com'),
+         jp.PathContainsPredicate('hostRules/hosts', 'host2.com')
+       ]
+     )
+    )
+    (contract_builder.new_clause_builder('Forwarding Rule Created',
+                                         retryable_for_secs=30)
+     .list_resource('globalForwardingRules')
+     .contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', self.__lb_name),
+         jp.PathContainsPredicate('portRange', '80-80')
+       ]
+     )
+    )
+    target_proxy_name = '%s-%s' % (self.__lb_name, HTTP_PROXY_PREFIX)
+    (contract_builder.new_clause_builder('Target Proxy Created',
+                                         retryable_for_secs=30)
+     .list_resource('targetHttpProxies')
+     .contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', target_proxy_name)
+       ]
+     )
+    )
+
+    return st.OperationContract(
+      self.new_post_operation(title='upsert full http lb',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+  def upsert_min_load_balancer(self):
+    '''Upserts a L7 LB with the minimum description.
+    '''
+    min_upsert = copy.deepcopy(self.__default_upsert)
```

Comment:
I think if default_upsert were just an object rather than a list then min_upsert would be dict(self.__default_upsert)